custom/icds_reports/views.py

```diff
@@ -1762,6 +1763,51 @@ def valid_state_names(self):
         return list(AwcLocation.objects.filter(aggregation_level=AggregationLevels.STATE, state_is_test=0).values_list('state_name', flat=True))
 
 
+@location_safe
+class DishaAPIViewV2(View):
+
+    def message(self, message_name):
+        state_names = ", ".join([state[0] for state in self.valid_states])
+        error_messages = {
+            "missing_date": "Please specify valid month and year",
+            "invalid_month": "Please specify a month that's older than a month and 5 days",
+            "invalid_state": "Please specify one of {} as state_name".format(state_names),
+        }
+        return {"message": error_messages[message_name]}
+
+    @method_decorator([api_auth, toggles.ICDS_DISHA_API_V2.required_decorator()])
+    def get(self, request, *args, **kwargs):
+        try:
+            month = int(request.GET.get('month'))
+            year = int(request.GET.get('year'))
```

Comment:
It might be worth adding a `cache_override` GET param, and bust the cache before returning response.

Reply:
yeah. I think it wont add much benefit because any way data is updated in 24 hours and caches timeout if put as 2 hours. So theoretically, it will be always be updated.

Reply:
Ha, make sense.

Reply:
Ha, make sense.